### PR TITLE
Ensure docker daemon does not depend on cryptfs logic on workers

### PR DESCRIPTION
**Pull request by rikukissa**
Data directories and cryptfs should only exist on manager node.

### Root cause

Docker service had been made dependant on CryptFS both in manager and worker machines. Worker machines do not hold any data, meaning they do not have cryptfs either. Event the shell scripts required to create the encryption do not exist on these machines.

### Fix

The logic is now so the dependency is only made between docker and cryptfs on manager nodes. I also made it so if this dependency had been created for a machine earlier, it is now removed.

## Checklist

- [ ] I have linked the correct Github issue under "Development"
- [ ] I have tested the changes locally, and written appropriate tests
- [ ] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [ ] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: infrastructure/server-setup/playbook.yml [hunk 1]
```diff
@@ -186,4 +185,18 @@
       tags:
         - traefik
 
+# Ensure incorrectly created override of docker daemon (1.9 beta) is removed on workers
```
Comment (makelicious) [inline infrastructure/server-setup/playbook.yml]: is this one off?
